ibm/service/contextbasedrestrictions/resource_ibm_cbr_zone_addresses.go

```diff
@@ -0,0 +1,295 @@
+			MinValueLength:             1,
+			MaxValueLength:             128,
+		},
+		validate.ValidateSchema{
+			Identifier:                 "x_correlation_id",
+			ValidateFunctionIdentifier: validate.ValidateRegexpLen,
+			Type:                       validate.TypeString,
+			Optional:                   true,
+			Regexp:                     `^[a-zA-Z0-9 ,\-_]+$`,
+			MinValueLength:             1,
+			MaxValueLength:             1024,
+		},
+		validate.ValidateSchema{
+			Identifier:                 "transaction_id",
+			ValidateFunctionIdentifier: validate.ValidateRegexpLen,
+			Type:                       validate.TypeString,
+			Optional:                   true,
+			Regexp:                     `^[a-zA-Z0-9 ,\-_]+$`,
+			MinValueLength:             1,
+			MaxValueLength:             1024,
+		},
+	)
+
+	resourceValidator := validate.ResourceValidator{ResourceName: "ibm_cbr_zone_addresses", Schema: validateSchema}
+	return &resourceValidator
+}
+
+func resourceIBMCbrZoneAddressesCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	addressId, _ := uuid.GenerateUUID()
+	d.SetId(composeZoneAddressesId(d.Get("zone_id").(string), "TF-"+addressId))
+	err := resourceReplaceZoneAddresses(context, d, meta, false)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func resourceIBMCbrZoneAddressesRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	contextBasedRestrictionsClient, err := meta.(conns.ClientSession).ContextBasedRestrictionsV1()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	zoneId, addressesId := decomposeZoneAddressesId(d.Id())
+
+	var zone *contextbasedrestrictionsv1.Zone
+	var found bool
+	zone, _, found, err = getZone(contextBasedRestrictionsClient, context, zoneId)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if !found {
+		d.SetId("")
+		return nil
+	}
+
+	if err = d.Set("zone_id", zoneId); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting zone_id: %s", err))
+	}
+
+	var addresses []map[string]interface{}
+	addresses, err = resourceDecodeAddressList(zone.Addresses, addressesId)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if len(addresses) == 0 {
+		d.SetId("")
+		return nil
+	}
+
+	if err = d.Set("addresses", addresses); err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting addresses: %s", err))
+	}
+
+	return nil
+}
+
+func resourceIBMCbrZoneAddressesUpdate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	err := resourceReplaceZoneAddresses(context, d, meta, false)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func resourceIBMCbrZoneAddressesDelete(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	err := d.Set("addresses", nil)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("Error setting addresses: %s", err))
+	}
+	err = resourceReplaceZoneAddresses(context, d, meta, true)
```

Comment:
What does resourceReplaceZoneAddresses do when its destroy.

Can we just do d.SetId("") and return if we don't have Delete specific API

Reply:
`resourceReplaceZoneAddresses` is called to remove any associated addresses from the zone before deleting the zone addresses TF resource.